Cargo.toml

```diff
@@ -40,6 +40,7 @@ sha2 = { version = "0.10.2", optional = true }
 sprintf = { version = "0.1.2", optional = true }
 parse-size = { version = "1.0.0", features = ["std"], optional = true }
 serde_yaml = { version = "0.8.26", optional = true }
+url = { version = "2.2.2", optional = true }
```

Comment:
In this PR you're only using stuff from `url::form_urlencoded`, which is a re-export from the `form_urlencoded` crate, so you could just use that crate directly instead